js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -895,35 +917,47 @@ const EditingController = modules.ViewController.inherit((function() {
             }
         },
 
-        _getInsertKey: function(parentKey) {
-            const that = this;
-            const dataController = that._dataController;
+        _createInsertInfo: function(parentKey) {
+            const dataController = this._dataController;
             const rows = dataController.items();
-            const editMode = getEditMode(that);
 
-            const insertKey = {
+            const insertInfo = {
                 parentKey,
                 pageIndex: dataController.pageIndex(),
-                rowIndex: that._getInsertRowIndex(parentKey)
+                rowIndex: this._getInsertRowIndex(parentKey)
             };
 
-            const row = rows[insertKey.rowIndex];
+            const row = rows[insertInfo.rowIndex];
 
             if(row && (!row.isEditing && row.rowType === 'detail' || row.rowType === 'detailAdaptive')) {
-                insertKey.rowIndex++;
+                insertInfo.rowIndex++;
             }
 
-            insertKey.dataRowIndex = dataController.getRowIndexOffset() + rows.filter(function(row, index) {
-                return index < insertKey.rowIndex && (row.rowType === 'data' && !row.isNewRow || row.rowType === 'group');
+            insertInfo.dataRowIndex = dataController.getRowIndexOffset() + rows.filter(function(row, index) {
+                return index < insertInfo.rowIndex && (row.rowType === 'data' && !row.isNewRow || row.rowType === 'group');
             }).length;
 
-            if(editMode !== EDIT_MODE_BATCH) {
-                this._setEditRowKey(insertKey, true);
+            insertInfo[INSERT_INDEX] = this._getInsertIndex();
+
+            return insertInfo;
+        },
+
+        _getInsertInfo: function({ parentKey, key } = {}) {
```

Comment:
Rename _getInsertInfo to _addInsertInfo because this method create insertInfo and add it to internal data.